### ISSUE TITLE
Documentation error: description of method "within"

### ISSUE THREAD

**Issue by NickGotts** [Toblerity/Shapely#1081]
The current description of method "within" is:

>   Returns ``True`` if the object's `boundary` and `interior` intersect only
>   with the `interior` of the other (not its `boundary` or `exterior`).

However, in fact the object's boundary can intersect the boundary of the other. For example:

> aa=Polygon([(0,0),(0,2),(2,2),(2,0)])
> bb=Polygon([(0,0),(0,1),(1,0)])
> bb.within(aa)

> True

If the current definition were correct, "within" would not be the inverse of "contains".

**Comment by Jeremiah-England** [comment]
Good catch! This is corroborated by the [GEOS docs](http://geos.refractions.net/ro/doxygen_docs/html/classgeos_1_1geom_1_1Geometry.html#a28) and a few other tests I ran (e.g. `geom.within(geom)` is `True`). 

[Link to Shapely docs](https://shapely.readthedocs.io/en/stable/manual.html#object.within)

Here's a first pass at some wording:

> Returns `True` if the object's *interior* intersects the *interior* of the other and *none* of the object intersects with the *exterior* of the other.
> 
> For example, `POINT (1 1)` is within `LINESTRING (0 0, 1 1, 2 2)`. However, `POINT (0 0)` is not because it only intersects the *boundary* of the LineString and not the *interior*. Intersecting with the *boundary* of the other is allowed though, `LINESTRING (0 0, 1 1)` is within `LINESTRING (0 0, 1 1, 2 2)` because it's *interior* does intersect the other's *interior* and none of it intersects the other's *exterior*. 
> 
> See also `covers_by` for a similar but slightly more inclusive relation. (...when those docs are merged.)

**Comment by NickGotts** [comment]
Thanks for the update!
Nick Gotts


> Closed #1081 as completed via #2392.
> 
> --
> Reply to this email directly or view it on
GitHub:
>
https://github.com/shapely/shapely/issues/1081#event-22377862501
>
You are receiving this because you authored the thread.
> 
> Message ID:
>
***@***.***>
